packages/core-backend/src/BackendWebSocketService.ts

```diff
@@ -0,0 +1,1316 @@
+  >;
+
+  readonly #enabledCallback: (() => boolean) | undefined;
+
+  #ws: WebSocket | undefined;
+
+  #state: WebSocketState = WebSocketState.DISCONNECTED;
+
+  #reconnectAttempts = 0;
+
+  #reconnectTimer: NodeJS.Timeout | null = null;
+
+  // Track the current connection promise to handle concurrent connection attempts
+  #connectionPromise: Promise<void> | null = null;
+
+  readonly #pendingRequests = new Map<
+    string,
+    {
+      resolve: (value: unknown) => void;
+      reject: (error: Error) => void;
+      timeout: NodeJS.Timeout;
+    }
+  >();
+
+  #connectedAt: number | null = null;
+
+  // Simplified subscription storage (single flat map)
+  // Key: subscription ID string (e.g., 'sub_abc123def456')
+  // Value: WebSocketSubscription object with channels, callback and metadata
+  readonly #subscriptions = new Map<string, WebSocketSubscription>();
+
+  // Channel-based callback storage
+  // Key: channel name (serves as unique identifier)
+  // Value: ChannelCallback configuration
+  readonly #channelCallbacks = new Map<string, ChannelCallback>();
+
+  // =============================================================================
+  // 1. CONSTRUCTOR & INITIALIZATION
+  // =============================================================================
+
+  /**
+   * Creates a new WebSocket service instance
+   *
+   * @param options - Configuration options for the WebSocket service
+   */
+  constructor(options: BackendWebSocketServiceOptions) {
+    this.#messenger = options.messenger;
+    this.#enabledCallback = options.enabledCallback;
+
+    this.#options = {
+      url: options.url,
+      timeout: options.timeout ?? 10000,
+      reconnectDelay: options.reconnectDelay ?? 500,
+      maxReconnectDelay: options.maxReconnectDelay ?? 5000,
+      requestTimeout: options.requestTimeout ?? 30000,
+    };
+
+    // Setup authentication (always enabled)
+    this.#setupAuthentication();
+
+    // Register action handlers using the method actions pattern
+    this.#messenger.registerMethodActionHandlers(
+      this,
+      MESSENGER_EXPOSED_METHODS,
+    );
+  }
+
+  /**
+   * Setup authentication event handling - simplified approach using AuthenticationController
+   * AuthenticationController.isSignedIn includes both wallet unlock AND identity provider auth.
+   * App lifecycle (AppStateWebSocketManager) handles WHEN to connect/disconnect for resources.
+   *
+   */
+  #setupAuthentication(): void {
+    try {
+      // Subscribe to authentication state changes - this includes wallet unlock state
+      // AuthenticationController can only be signed in if wallet is unlocked
+      // Using selector to only listen for isSignedIn property changes for better performance
+      this.#messenger.subscribe(
+        'AuthenticationController:stateChange',
+        (isSignedIn: boolean) => {
+          if (isSignedIn) {
+            // User signed in (wallet unlocked + authenticated) - try to connect
+            console.debug(
```

Comment:
Can we do a sweep through the console.log/debug lines?
Lets see if these need to be kept (for debugging purposes) or can be removed/cleaned up.